src/utils/azure_storage.ts

```diff
@@ -186,6 +187,35 @@ export const getBlobAsText = (
     );
   });
 
+/**
+ * Get a blob content as text (string). In case of error, it will return error details.
+ *
+ * @param blobService     the Azure blob service
+ * @param containerName   the name of the Azure blob storage container
+ * @param blobName        blob file name
+ */
+export const getBlobAsTextWithError = (
+  blobService: azureStorage.BlobService,
+  containerName: string,
+  options: azureStorage.BlobService.GetBlobRequestOptions = {}
+) => (
+  blobName: string
+): TE.TaskEither<azureStorage.StorageError, Option<string>> =>
+  pipe(
+    new Promise<E.Either<azureStorage.StorageError, Option<string>>>(
+      resolve => {
+        blobService.getBlobToText(
+          containerName,
+          blobName,
+          options,
+          (err, result, _) =>
+            err ? resolve(E.left(err)) : resolve(E.right(fromNullable(result)))
+        );
+      }
+    ),
+    constant
```

Comment:
Why not using `TE.tryCatch` here?

Reply:
The tryCatch will route a throwed error in the left side. In this case we had a callback managing the error.
How could we use the tryCatch to manage it?

PS: I tried with TE.taskify, but I could not understand how to make it work.